Makefile

```diff
@@ -286,8 +355,20 @@ all: all-c-tests all-cpp-tests all-cc-tests
 INCLUDE = -I ./$(L3_INCDIR)
 INCLUDE += -I ./$(dir $<)
 
+# use += here, so that extra flags can be provided via the environment
+ifeq "$(BUILD_MODE)" "debug"
+    CFLAGS += -DDEBUG
```

Comment:
@gregthelaw -- Fyi, so you are alerted by this change I sneaked-in as part of the review cycle.

For debug builds, we will now issue `-DDEBUG` in `CFLAGS`.

That will be checked-for in `l3.c`, to assert under DEBUG mode that `assert(loc == 0);`

This ties-in to the comment you made that we should rely on the fact that `mmap()` returns zero-filled memory regions.